### ISSUE TITLE
Bug: Add Docker Template button does not work in newer versions of Jenkins

### ISSUE THREAD

**Issue by samrocketman** [KostyaSha/yet-another-docker-plugin#136]
# What doesn't work

In newer versions of Jenkins (tested using LTS 2.32.2 and weekly 2.50), the Yet Another Docker Plugin can't add Docker Templates to its configuration.

# Steps to reproduce

1. Set a fresh `$JENKINS_HOME`
1. Run `java -jar jenkins.war` with latest Jenkins LTS or weekly.
2. Install YADP and restart Jenkins.
3. Visit `Manage Jenkins > Configure System`
4. Scroll to the bottom and under `Clouds` add `Yet Another Docker`.
5. Under Images in the configuration click the dropdown named `Add Docker Template`.

What was expected to happen?  I was expecting to be able to add a docker template for provisioning.

What actually happened?  Nothing, it wouldn't show a dropdown at all or try to add a new template.

# Environment

```
Ubuntu 16.04.2 LTS
Linux 4.4.0-65-generic x86_64
java version "1.8.0_112"
Java(TM) SE Runtime Environment (build 1.8.0_112-b15)
Java HotSpot(TM) 64-Bit Server VM (build 25.112-b15, mixed mode)
```

Browsers tested:
* Google Chrome Version 57.0.2987.98 (64-bit)
* Firefox 45.0.2

Jenkins versions tested:

* Jenkins ver. 2.32.2 (LTS)
* Jenkins ver. 2.50 (weekly)

**Comment by fboudra** [comment]
I can confirm the issue. Running LTS, upgraded to YAD 0.1.0-rc33. It works fine on another LTS instance using YAD 0.1.0-rc30.

**Comment by KostyaSha** [comment]
Do you have any stacktraces in global settings?

**Comment by samrocketman** [comment]
Unfortunately, I paid close attention to the global logs.  There's no stack traces.

I used Firebug in Firefox to try to debug the UI and there's not much I could see in the way of what exactly the root cause was.

**Comment by samrocketman** [comment]
@fboudra I forgot to say _what version_ of YADP.  I was using the same version noted in @fboudra's comment.

**Comment by samrocketman** [comment]
I confirm YADP 0.1.0-rc31 works as well.  I haven't tested 0.1.0-rc32.

**Comment by KostyaSha** [comment]
Will check, i made some refactorings, but i had unchanged config round-trip tests...

**Comment by KostyaSha** [comment]
`repeatableHeteroProperty` is broken

**Comment by KostyaSha** [comment]
well, it render existing templates, so getter works, but it doesn't show dropdown. I have no ideas how to debug it. I did some refactorings, but all fields on the same place.

**Comment by KostyaSha** [comment]
Specifying descriptor helps, but then it can't bind. so issues somewhere in bindings and types or fields..

**Comment by JelteF** [comment]
Running into this as well. How can I get the older working version 0.1.0-rc31?

**Comment by winnerwbx** [comment]
Temporary workaround: Edit ${JENKINS_HOME}/config.xml directly. 
PS: you must generate a guid by youself in <id> tag.

**Comment by jwtodd** [comment]
firstly, thx for furthering the jenkins-docker-plugin work ... saying this after doing a docker u/g in order to try out the new 'docker stack ...' feature, which in turn horked my docker-plugin install. after searching around a bit, i happened upon this plugin and it works with the docker u/g as expected.

secondarily, i can confirm the inability to add a template via the ui as described above for 0.1.0-rc33 but the work around described by @winnerwbx works as described. the work around described is perfect for my env in that jenkins-master is run from within a docker container like that of the slaves and the config files are deployed into the container with macros replacing the key ids, credentials, etc.

so basically ... thx :)

and i am looking forward to an u/g that resolves the above.

one question: to enable slaves to be instantiated via jnlp, i assume that requires a jenkins stack to be resident in the slave image? presently i just have jdk/gradle/etc and connect via ssh.

**Comment by KostyaSha** [comment]
found, seems fixed.
